File: programmers/new/level2/기능개발.java
import java.io.*;
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> list = new ArrayList<Integer>();
        int[] arr = new int[progresses.length];
        for(int i = 0; i < progresses.length; ++i) {
        	arr[i] = (100-progresses[i])/speeds[i];
        	if((100-progresses[i])%speeds[i] > 0) {
        		arr[i]++;
        	}
        }
        int current = arr[0];
        int count = 1;
        for(int i = 1; i < arr.length; ++i) {
        	if(arr[i] <= current) {
        		count++;
        	}else {
        		list.add(count);
        		current = arr[i];
        		count = 1;
        	}
        }
        list.add(count);
        int[] answer = new int[list.size()];
        for(int i = 0; i < answer.length; ++i) {
        	answer[i] = list.get(i);
        }
        return answer;
    }
}

public class 기능개발 {
	public static void main(String[] args) throws Exception{
		int[] progresses = {93,30,55};
		int[] speeds = {1,30,5};
		Solution sol = new Solution();
		System.out.println(Arrays.toString(sol.solution(progresses, speeds)));
	}
}

File: programmers/js/new/level2/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    const needs = [];
    for (let i = 0; i < progresses.length; ++i) {
        needs.push(Math.ceil((100 - progresses[i]) / speeds[i]));
    }
    let stack = [];
    let sMax = 0;
    needs.forEach((e) => {
        if (stack.length === 0) {
            stack.push(e);
            sMax = e;
        } else {
            const top = stack[stack.length - 1];
            if (e <= sMax) {
                stack.push(e);
            } else {
                answer.push(stack.length);
                stack = [e];
                sMax = e;
            }
        }
    });
    answer.push(stack.length);
    return answer;
}
